File: system-test/dotnet/Dsl.Core/Scenario/Given/Steps/GivenClock.cs
using Dsl.Port.Given.Steps;
using Driver.Adapter;
using Dsl.Core.Gherkin;
using static Dsl.Core.Gherkin.GherkinDefaults;

namespace Dsl.Core.Scenario.Given;

public class GivenClock : BaseGiven, IGivenClock
{
    private string? _time;

    public GivenClock(GivenStage givenClause) : base(givenClause)
    {
        WithTime(DefaultTime);
    }

    public GivenClock WithTime(string? time)
    {
        _time = time;
        return this;
    }

    IGivenClock IGivenClock.WithTime(string? time) => WithTime(time);

    internal override async Task Execute(UseCaseDsl app)
    {
        (await app.Clock().ReturnsTime()
            .Time(_time)
            .Execute())
            .ShouldSucceed();
    }
}

File: system-test/java/src/main/java/com/optivem/eshop/dsl/port/given/steps/GivenClock.java
package com.optivem.eshop.dsl.port.given.steps;

import com.optivem.eshop.dsl.port.given.steps.base.GivenStep;

public interface GivenClock extends GivenStep {
    GivenClock withTime();
    GivenClock withTime(String time);
    GivenClock withWeekday();
    GivenClock withWeekend();
}

